syntax = "proto3";
import "google/protobuf/empty.proto";

package gl;


message Point3 {
    float x = 1;
    float y = 2;
    float z = 3;
    optional float r = 4;
    optional float g = 5;
    optional float b = 6;
}

message PointCloud3 { 
    repeated Point3 points = 1;
    optional string entity_name = 2;
    optional float point_size = 3;
}

message Line {
    float x0 = 1;
    float y0 = 2;
    float z0 = 3;

    float x1 = 4;
    float y1 = 5;
    float z1 = 6;

    optional string entity_name = 7;
}

// All lines share the same color
message LinesRequest {
    repeated Line lines = 1;
    float r = 2;
    float g = 3;
    float b = 4;
}


service addToScene {
    rpc addPoint(Point3) returns (google.protobuf.Empty);
    rpc addPointCloud(PointCloud3) returns (google.protobuf.Empty);
    rpc addLines(LinesRequest) returns (google.protobuf.Empty);

    rpc streamPointClouds(stream PointCloud3) returns (google.protobuf.Empty);

    rpc resetScene(google.protobuf.Empty) returns (google.protobuf.Empty);
}